src/KafkaSchemaRegistryApiApiClient.php

```diff
@@ -0,0 +1,229 @@
+                ->call(
+                    'POST',
+                    sprintf('subjects/%s/versions', $subjectName),
+                    $this->createRequestBodyFromSchema($schema)
+                ) ?? [];
+    }
+
+    /**
+     * @param string $schema
+     * @param string $subjectName
+     * @param string $version
+     * @return bool
+     * @throws Exception
+     */
+    public function checkSchemaCompatibilityForVersion(
+        string $subjectName,
+        string $schema,
+        string $version = self::VERSION_LATEST
+    ): bool {
+        try {
+            $results = $this
+                   ->httpClient
+                   ->call(
+                       'POST',
+                       sprintf('compatibility/subjects/%s/versions/%s', $subjectName, $version),
+                       $this->createRequestBodyFromSchema($schema)
+                   ) ?? [];
+        } catch (SubjectNotFoundException $e) {
+            return true;
+        }
+
+        /** @var bool $result */
+        $result = $results['is_compatible'];
+        return $result;
+    }
+
+    /**
+     * @param string $subjectName
+     * @return string|null
+     * @throws Exception
+     */
+    public function getSubjectCompatibilityLevel(string $subjectName): ?string
+    {
+        try {
+            $results = $this->httpClient->call('GET', sprintf('config/%s', $subjectName));
+            return $results['compatibilityLevel'];
+        } catch (SubjectNotFoundException $e) {
+            return $this->getDefaultCompatibilityLevel();
+        }
+    }
+
+    /**
+     * @param string $subjectName
+     * @param string $level
+     * @return bool
+     */
+    public function setSubjectCompatibilityLevel(string $subjectName, string $level = self::LEVEL_FULL): bool
+    {
+        $this->httpClient->call('PUT', sprintf('config/%s', $subjectName), ['compatibility' => $level]);
+        return true;
+    }
+
+    /**
+     * @return string
+     */
+    public function getDefaultCompatibilityLevel(): string
+    {
+        $results = $this->httpClient->call('GET', 'config');
+        return $results['compatibilityLevel'];
+    }
+
+    /**
+     * @param string $level
+     * @return bool
+     */
+    public function setDefaultCompatibilityLevel(string $level = self::LEVEL_FULL): bool
+    {
+        $this->httpClient->call('PUT', 'config', ['compatibility' => $level]);
+        return true;
+    }
+
+    /**
+     * @param string $subjectName
+     * @param string $schema
+     * @return string|null
+     */
+    public function getVersionForSchema(string $subjectName, string $schema): ?string
+    {
+        try {
+            $results = $this
+                    ->httpClient
+                    ->call(
+                        'POST',
+                        sprintf('subjects/%s', $subjectName),
+                        $this->createRequestBodyFromSchema($schema)
+                    ) ?? [];
+
+            return (string) $results['version'];
```

Comment:
is there a reason you cast it to string?

Reply:
When we get version, they accept string, but give us integer, so I wanted to be same.

Reply:
is that ok?